pandas/core/reshape/pivot.py

```diff
@@ -70,10 +70,39 @@ def pivot_table(
                 margins_name=margins_name,
                 observed=observed,
             )
-            pieces.append(table)
+            pieces.append(_table)
             keys.append(getattr(func, "__name__", func))
 
-        return concat(pieces, keys=keys, axis=1)
+        table = concat(pieces, keys=keys, axis=1)
+        return table.__finalize__(data, method="pivot_table")
+
+    table = _pivot_table(
```

Comment:
this is way more complicated to read. need to simplify this or have a better name for _pivot_table -> __internal_pivot_table, and add a doc-string & types.

Reply:
this maybe worth reverting and doing in a dedicated PR.

Reply:
Just finished reading the requirements for doc-string. Will rename the function and add a doc-string & types.